tests/test_api_csv.py

```diff
@@ -33,96 +33,134 @@ def test_csv_extract_bbox_and_tbox_semicolon_delimiter():
-    assert result["tbox"] == ['01.08.2017', '30.09.2019']
+    assert result["tbox"] == ['2017-08-01', '2019-09-30']
 
 def test_csv_extract_bbox_and_tbox_with_cols_diff_order_and_alt_names2():
     result = geoextent.fromFile('tests/testdata/csv/cities_NL_case4.csv', bbox=True, tbox=True)
     assert "bbox" in result
     assert "tbox" in result
     assert result["bbox"] == [4.3175, 51.434444, 6.574722, 53.217222]
-    assert result["tbox"] == ['01.08.2017', '30.09.2019']
+    assert result["tbox"] == ['2017-08-01', '2019-09-30']
 
 def test_csv_extract_bbox_and_tbox_with_cols_diff_order_and_alt_names3():
     result = geoextent.fromFile('tests/testdata/csv/cities_NL_case5.csv', bbox=True, tbox=True)
     assert "bbox" in result
     assert "tbox" in result
     assert result["bbox"] == [4.3175, 51.434444, 6.574722, 53.217222]
-    assert result["tbox"] == ['01.08.2017', '30.09.2019']
+    assert result["tbox"] == ['2017-08-01', '2019-09-30']
 
+def test_csv_extract_tbox_ISO8601_time_format():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS.csv', bbox=False, tbox = True)
+    assert "bbox" not in result
+    assert "tbox" in result
+    assert result["tbox"] == ['2017-04-08', '2020-02-06']
+
+def test_csv_extract_tbox_DD_MM_YYYY_time_format():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS_dd_mm_yyyy.csv', bbox=False, tbox = True)
+    assert "bbox" not in result
+    assert "tbox" in result
+    assert result["tbox"] == ['2017-04-19', '2018-01-31']
+
+def test_csv_extract_tbox_month_abbr_dd_yyyy_time_formats():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS_month_abbr_dd_yyyy_time_format.csv', bbox=False, tbox = True)
+    assert "bbox" not in result
+    assert result["tbox"] == ['2017-04-09', '2017-07-20']
+
+def test_csv_extract_tbox_two_diff_time_formats():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS_mixed_time_formats.csv', bbox=False, tbox = True)
+    assert "bbox" not in result
+    assert "tbox" not in result
+
+def test_csv_extract_tbox_random_sample():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS.csv', bbox=False, tbox=True , num_sample = 5)
+    assert "bbox" not in result
+    assert "tbox" in result
+    assert result["tbox"] == ['2017-04-08', '2020-02-06']
+
+def test_csv_extract_tbox_random_sample_invalid():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS.csv', bbox=False, tbox=True , num_sample = -1)
+    assert "bbox" not in result
+    assert "tbox" not in result
+
+def test_csv_extract_tbox_random_sample_value_larger_than_data():
+    result = geoextent.fromFile('tests/testdata/csv/3DCMTcatalog_TakemuraEPS.csv', bbox=False, tbox=True , num_sample = 1000000)
+    assert "bbox" not in result
+    assert "tbox" in result
+    assert result["tbox"] == ['2017-04-08', '2020-02-06']
```

Comment:
Please document the test data source in https://github.com/o2r-project/geoextent/blob/master/tests/testdata/data-sources.md